third_party/terraform/resources/resource_bigtable_instance.go

```diff
@@ -185,22 +189,31 @@ func resourceBigtableInstanceRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("project", project)
 
-	clusters := d.Get("cluster").([]interface{})
-	clusterState := []map[string]interface{}{}
-	for _, cl := range clusters {
-		cluster := cl.(map[string]interface{})
-		clus, err := c.GetCluster(ctx, instance.Name, cluster["cluster_id"].(string))
-		if err != nil {
-			if isGoogleApiErrorWithCode(err, 404) {
-				log.Printf("[WARN] Cluster %q not found, not setting it in state", cluster["cluster_id"].(string))
-				continue
+	clusters, err := c.Clusters(ctx, instance.Name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving instance clusters. %s", err)
+	}
+
+	clustersNewState := []map[string]interface{}{}
+	for i, cluster := range clusters {
+		// DEVELOPMENT clusters have num_nodes = 0 on their first (and only)
+		// cluster while PRODUCTION clusters will have at least 3.
+		if i == 0 {
+			var instanceType string
+			if cluster.ServeNodes == 0 {
+				instanceType = "DEVELOPMENT"
+			} else {
+				instanceType = "PRODUCTION"
 			}
-			return fmt.Errorf("Error retrieving cluster %q: %s", cluster["cluster_id"].(string), err.Error())
+
+			d.Set("instance_type", instanceType)
 		}
-		clusterState = append(clusterState, flattenBigtableCluster(clus, cluster["storage_type"].(string)))
+
+		clustersNewState = append(clustersNewState, flattenBigtableCluster(cluster))
 	}
 
-	err = d.Set("cluster", clusterState)
+	log.Printf("[DEBUG] Setting clusters in state: %#v", clustersNewState)
+	err = d.Set("cluster", clustersNewState)
```

Comment:
We keep clusters in state but they are separately managed resources?

Reply:
Yep- this resource didn't have clusters originally, and they were only added later in the service's lifecycle. Cluster conf was inline on the resource. For compatibility reasons, we broke it into a block in-resource still and it's remained like that.

We can't use the GKE or Network/Subnetwork model because instances require 1+ clusters at all times.